openedx/event-bus-redis
# feat: limit stream length by using redis stream trim option

**Description:** Implements `trim` option as decided in ADR: https://github.com/openedx/event-bus-redis/blob/main/docs/decisions/0003-limiting-stream-length.rst

Also, adds more information to README.

**JIRA:** `Private-ref`: https://tasks.opencraft.com/browse/BB-7451

**Dependencies:** None

**Merge deadline:** List merge deadline (if any)

**Testing instructions:**

1. Make sure python requirements are updated in your virtual env using `make requirements`.
2. Start redis using `make redis_up`.
3. Open redis cli in one terminal using `make redis_shell`.
4. `EVENT_BUS_REDIS_STREAM_MAX_LEN` is set to 5 for testing in Makefile command: `produce_test_event`.
5. Run `make produce_test_event` once or twice and check length of stream in redis cli window using `XLEN dev-xblock-deleted` command.
6. Now run `for run in {1..10}; do make produce_test_event; done` to produce 10 events. Keep checking stream length while producing. Since the trimming is not done exactly at 5 events, it might be required to produce more events to see trimming in action. At some point the stream length should be reset to around 5.

**Reviewers:**
- [ ] @tecoholic
- [ ] @bmtcril 

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
